dream_demo.ipynb

```diff
@@ -0,0 +1,261 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import dream\n",
+    "import powder_reduction\n",
+    "import scipp as sc\n",
+    "from scipp import Dim"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Testing dream class that generates fake powder dataset\n",
+    "<div class=\"alert alert-block alert-info\">\n",
+    "<b>Information:</b> This notebook shows how to use some of `scipp` functionalities related to the \"DREAM\" powder diffraction reduction workflow.\n",
+    "\n",
+    "<b>Requirements:</b> To run this notebook, you need `matplotlib`, `scipp`, `scipy` installed as well as the Python scripts `powder_reduction.py` and `dream.py` placed in the same folder as this notebook.   \n",
+    "</div>\n",
+    "\n",
+    "- Uses the `dream` module to create a fake sparse dataset with a given number of pixels and events.\n",
+    "\n",
+    "- The dataset includes a calibration file `dream.cal` which is used to convert the resulting signal to d-spacing.\n",
+    "\n",
+    "- At the end the resulting data is plotted using `plotly`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Script for demonstration of event generation\n",
+    "n_pixel = int(2E4)\n",
+    "n_events = int(2E6)\n",
+    "\n",
+    "Dream = dream.DreamTest(n_pixel) # Create a fake dream powder diffractometer with detector positions and calibration data\n",
```

Comment:
Uppercase names are usually only used for classes, not for variables (see https://www.python.org/dev/peps/pep-0008/). Reading `Dream`, most people would think it is a class, but it is not.

Reply:
The issue arises because the name can not be the same as the module name, as the cell in the notebook can not be rerun when that is the case.

Reply:
You will need to find a different name, or import the module under a different name (`import dream as xxx`). As soon as some `autopep8` check such as `flake8` (which we have in the `scipp` repository) is introduced, this will fail. Also, we should not teach bad style to users.